csharp/src/Ice/RemoteException.cs

```diff
@@ -133,4 +149,51 @@ public static class RemoteExceptionExtensions
         /// <returns>A SlicedData value that provides the list of sliced-off slices.</returns>
         public static SlicedData? GetSlicedData(this RemoteException ex) => ex.SlicedData;
     }
+
+    public partial class ObjectNotExistException
+    {
+        /// <inheritdoc/>
+        protected override string DefaultMessage =>
+            $"could not find servant for Ice object `{Origin!.Value.Identity}'" +
+            (Origin.Value.Facet.Length > 0 ? $" with facet `{Origin.Value.Facet}'" : "") +
+            $" while attempting to dispatch operation `{Origin.Value.Operation}'";
+    }
+
+    public partial class OperationNotExistException
+    {
+        /// <inheritdoc/>
+        protected override string DefaultMessage =>
+            $"could not find operation `{Origin!.Value.Operation}' for Ice object `{Origin.Value.Identity}'" +
+            (Origin.Value.Facet.Length > 0 ? $" with facet `{Origin.Value.Facet}'" : "");
+    }
+
+    public partial class UnhandledException
+    {
+        /// <summary>Constructs a new exception.</summary>
+        /// <param name="identity">The Ice object Identity.</param>
+        /// <param name="facet">The Ice object facet.</param>
+        /// <param name="operation">The operation name.</param>
+        /// <param name="innerException">The exception that is the cause of the current exception.</param>
+        public UnhandledException(Identity identity, string facet, string operation, Exception innerException)
```

Comment:
We should rework this ctor. It's odd to pass these parameters only for this UnhandledException.

Reply:
Should we pass current and also add a RemoteExceptionOrigin that takes Current object?

Reply:
I think we can rework this ctor to just receive the Current object, or should we create a RemoteExceptionOrigin and use it as the parameter?